docs/DeveloperGuide.adoc

```diff
@@ -0,0 +1,665 @@
+class to print out the search results.
+
+The following sequence diagram summarizes how the `SearchCommand` and `SearchdCommand` works:
+
+image::search.png[Search operations]
+
+==== Design Considerations:
+* Creating 2 separate classes for `SearchCommand` and `SearchdCommand`
+** Rationale: +
+To create 2 separate commands so that users can filter their search query more easily.
+** Alternatives Considered: +
+1. Use a `Search` class that implements both functions of `SearchCommand` and `SearchdCommand`
+*** Pros: Reduced coupling. Improved code structure.
+*** Cons: More difficult to implement
+2. Create another `SearchdCommand` within the `Parser` class that does the same operations as the `SearchdCommand`.
+*** Pros: Easier to implement.
+*** Cons: Makes the code for `Parser` unnecessarily long. Makes the code less OOP.
+
+=== Clear Task feature
+==== Current Implementation
+The `ClearCommand` inherits from the `Command` class and initializes the `clearParam` to check which clear function
+has to be executed
+
+Given below is an example usage of the `clear all` command:
+
+*Step 1* +
+The user launches the app and retrieves the tasks which are saved under a local file using `Storage`.
+
+*Step 2* +
+The user enters `clear all` into the command line. Method `parseCommand()` from the `Parser` class will be called to
+parse the command provided.
+
+*Step 3* +
+A new instance of `ClearCommand` with `clearParam` initialized will be created. The `execute()` method of
+`DeleteCommand` will then be called.
+
+*Step 4* +
+The `execute()` method will then call the `clearAll()` method in the `ClearCommand` class :
+
+* If there are no tasks in the existing task list, it will initialize a new `CommandResult` class that prints out an error
+message indicating an empty task list
+* If there are tasks in the existing task list, it will call the `clearList()` method from the `TaskList` class to clear the
+existing taskList
+
+Given below is an example usage of `clear done` command:
+
+*Step 1* +
+The user launches the app and retrieves the tasks which are saved under a local file using `Storage`.
+
+*Step 2* +
+The user enters `clear all` into the command line. Method `parseCommand()` from the `Parser` class will be called to
+parse the command provided.
+
+*Step 3* +
+A new instance of `ClearCommand` with `clearParam` initialized will be created. The `execute()` method of
+`DeleteCommand` will then be called.
+
+*Step 4* +
+The `execute()` method will then call the `clearDone()` method in the `ClearCommand` class :
+
+* If there are no tasks in the existing task list, it will initialize a new `CommandResult` class that prints out an error
+message indicating an empty task list
+* If there are tasks in the existing task list, it will call the `clearDone()` method that will call the `deleteAllDone()`
+method in the `TaskList` class
+
+The following sequence diagram summarizes how the `ClearCommand` operation works: +
+
+image::clear.png[clear command]
```

Comment:
1. Is this correct? The control seems to change from TaskCommand to CommandResults even before the control reaches the function.

2. If the CommandResult class is used for the 3 different alternative branches, shouldn't it be constructed/deleted outside the alt block? If not the class needs to initialized/deleted in all three branches separately.

3. Shouldn't the self invocation arrows point to the top of the activation bar?

![Annotation 2020-03-31 123402](https://user-images.githubusercontent.com/59698024/77988313-9f91d780-734e-11ea-944d-b9543b7a4d05.jpg)